pkg/cloud/services/ec2/instances.go

```diff
@@ -1008,6 +1022,56 @@ func (s *Service) ModifyInstanceMetadataOptions(instanceID string, options *infr
 	return nil
 }
 
+// GetDHCPOptionSetDomainName returns the domain DNS name for the VPC from the DHCP Options.
+func (s *Service) GetDHCPOptionSetDomainName(ec2client ec2iface.EC2API, vpcID *string) *string {
+	fmt.Println("vpcID", vpcID)
+	log := s.scope.GetLogger()
+
+	if vpcID == nil {
+		log.Error(errors.New("vpcID is nil"), "vpcID is nil")
+		return nil
+	}
+
+	vpcInput := &ec2.DescribeVpcsInput{
+		VpcIds: []*string{vpcID},
+	}
+
+	vpcResult, err := ec2client.DescribeVpcs(vpcInput)
+	if err != nil {
+		log.Error(err, "failed to describe VPC", "vpcID", *vpcID)
+		return nil
+	}
+
+	dhcpInput := &ec2.DescribeDhcpOptionsInput{
+		DhcpOptionsIds: []*string{vpcResult.Vpcs[0].DhcpOptionsId},
+	}
+
+	log.Info("Fetching DHCP Options to get the domain-name", "dhcpOptionsID", *vpcResult.Vpcs[0].DhcpOptionsId)
+
+	dhcpResult, err := ec2client.DescribeDhcpOptions(dhcpInput)
+	if err != nil {
+		log.Error(err, "failed to describe DHCP Options Set", "DhcpOptionsSet", *dhcpResult)
```

Comment:
```suggestion
		log.Info(err, "failed to describe DHCP Options Set, skipping DHCP Option Set discovery", "DhcpOptionsSet", vpcResult.Vpcs[0].DhcpOptionsId)
```
`dhcpResult` is effectively guaranteed to be nil if an error is returned here, also matching earlier log prints to add context around what is being skipped